velox/vector/tests/VectorMaker.cpp

```diff
@@ -220,6 +220,39 @@ ArrayVectorPtr VectorMaker::arrayOfRowVector(
       0);
 }
 
+ArrayVectorPtr VectorMaker::createArrayVector(
+    const std::vector<vector_size_t>& offsets,
+    const std::vector<vector_size_t>& sizes,
+    const VectorPtr elementVector,
+    const std::vector<vector_size_t>& nulls) {
+  BufferPtr offsetsBuffer = allocateOffsets(offsets.size(), pool_);
+  BufferPtr sizesBuffer = allocateSizes(sizes.size(), pool_);
+  BufferPtr nullsBuffer =
```

Comment:
nullsBuffer needs to be allocated only if nulls are not empty, hence, the allocation can be moved inside the check below. Similarly, rawNulls variable can go inside that "if" as well.